sklearn/setup.py

```diff
@@ -1,8 +1,36 @@
+from pathlib import Path
 import os
 
 from sklearn._build_utils import maybe_cythonize_extensions
 
 
+_DEPRECATED_MODULES = {
```

Comment:
Maybe put this whole logic in another file so we can just mark it to be removed in 0.24